Sources/Networking/SessionTaskService.swift

```diff
@@ -124,7 +124,9 @@ open class SessionTaskService {
         downloadTask(withResumeData: resumeData) { result in
             switch result {
             case .success(let data):
-                success(data)
+                DispatchQueue.main.async {
```

Comment:
Why did you add it here? Looks like it extra, because you already wrap to main tread callbacks in `RequestService`